pkgs/applications/networking/instant-messengers/discord/linux.nix

```diff
@@ -88,132 +89,164 @@ let
-      echo 'require("${moonlight}/injector.js").inject(require("path").join(__dirname, "../_app.asar"));' > $out/opt/${binaryName}/resources/app/injector.js
+  # FHS Environment with all necessary libraries
+  fhsEnv = buildFHSEnv {
+    name = "${pname}-fhs";
+    
+    targetPkgs = pkgs: [
+      # Core system libraries
+      pkgs.stdenv.cc.cc.lib
+      pkgs.glibc
+      
+      # Graphics and display
+      pkgs.libglvnd
+      pkgs.mesa
+      pkgs.libdrm
+      pkgs.libgbm
+      pkgs.wayland
+      
+      # X11 libraries
+      pkgs.xorg.libX11
+      pkgs.xorg.libXcomposite
+      pkgs.xorg.libXcursor
+      pkgs.xorg.libXdamage
+      pkgs.xorg.libXext
+      pkgs.xorg.libXfixes
+      pkgs.xorg.libXi
+      pkgs.xorg.libXrandr
+      pkgs.xorg.libXrender
+      pkgs.xorg.libXtst
+      pkgs.xorg.libXScrnSaver
+      pkgs.xorg.libxcb
+      pkgs.xorg.libxshmfence
+      
+      # GTK and desktop integration
+      pkgs.gtk3
+      pkgs.glib
+      pkgs.cairo
+      pkgs.pango
+      pkgs.gdk-pixbuf
+      pkgs.atk
+      pkgs.at-spi2-atk
+      pkgs.at-spi2-core
+      pkgs.libappindicator-gtk3
+      pkgs.libdbusmenu
+      
+      # Audio
+      pkgs.alsa-lib
+      pkgs.libpulseaudio
+      pkgs.pipewire
+      
+      # Other system libraries
+      pkgs.dbus
+      pkgs.systemd
+      pkgs.fontconfig
+      pkgs.freetype
+      pkgs.expat
+      pkgs.libuuid
+      pkgs.cups
+      pkgs.nspr
+      pkgs.nss
+      pkgs.libnotify
+      pkgs.libcxx
+      pkgs.libxkbcommon
+    ] ++ lib.optional withTTS pkgs.speechd-minimal;
+    
+    multiPkgs = pkgs: [
+      # Additional 32-bit libraries if needed
+      pkgs.alsa-lib
+      pkgs.libpulseaudio
+    ];
+    
+    # Set up the runtime environment
+    runScript = writeScript "${pname}-wrapper" ''
+      #!/bin/bash
+      
+      # Run the disable updates script if enabled
+      ${lib.optionalString disableUpdates ''
+        ${lib.getExe disableBreakingUpdates}
+      ''}
+      
+      # Set up environment variables
+      export XDG_DATA_DIRS="${gtk3}/share/gsettings-schemas/${gtk3.name}/:$XDG_DATA_DIRS"
+      
+      # Handle Wayland
+      if [[ -n "$NIXOS_OZONE_WL" && -n "$WAYLAND_DISPLAY" ]]; then
+        WAYLAND_FLAGS="--ozone-platform=wayland --enable-features=WaylandWindowDecorations --enable-wayland-ime=true"
+      fi
+      
+      # Handle TTS
+      ${lib.optionalString withTTS ''
+        if [[ "''${NIXOS_SPEECH:-default}" != "False" ]]; then
+          export NIXOS_SPEECH=True
+          TTS_FLAGS="--enable-speech-dispatcher"
+        fi
+      ''}
+      
+      # Handle autoscroll
+      ${lib.optionalString enableAutoscroll ''
+        AUTOSCROLL_FLAGS="--enable-blink-features=MiddleClickAutoscroll"
+      ''}
+      
+      # Execute Discord
+      exec ${discordDir}/${binaryName} \
+        --no-sandbox \
+        --disable-gpu-sandbox \
```

Comment:
Yeah, this definitely seems non-ideal lol. Hopefully there's some other way around the zygote/GPU launch issues?

Reply:
So this seems to be common between other electron apps, and the only other solution I've found seems to be not putting it in a sandbox, or using something like the Flathub version does with https://github.com/refi64/zypak to wrap Chromium in a way that it works in a nested sandbox. I don't see any specific usages of this tool in nixpkgs but perhaps there's another option.

At this point though this works well enough for my use cases so I won't personally be looking into other paths